pkg/controller/resource_sync_task.go

```diff
@@ -289,23 +296,67 @@ func (st *resourceSyncTask) prevalidate(res *smith_v1.Resource) error {
 				return errors.WithStack(err)
 			}
 			serviceInstance := actual.(*sc_v1b1.ServiceInstance)
```

Comment:
Scheme's conversion methods do not guarantee that a deep copy will be made. And looks like this method mutates the object - should deep copy it first.

Reply:
Argh, this is really annoying. I wish it was simpler for SpecProcessor to just do a deep copy automatically (guess it would have to use reflection?).

Reply:
Spec processor mutates the object - this is by design. But in this particular case we don't want that so need to make a copy. No need for reflectoion, there is a helper method to do "json" copying. `k8s.io/apimachinery/pkg/runtime/DeepCopyJSON()` and `DeepCopyJSONValue()`.

Reply:
I fixed it to use the deepcopy above before I posted that ;) - it should be fine now. I'm just leery of things that mutate an object in place (I would prefer that SpecProcessor did that by default, but AFAIK DeepCopyJSON would not be sufficient since it could have typed objects, no?)